tests/test_callback.py

```diff
@@ -30,23 +34,54 @@ def run_playbook_callback(tmpdir):
     return run_playbook(playbook, inventory=inventory, extra_env=extra_env)
 
 
-def test_callback(tmpdir, vcrmode):
-    run = run_playbook_callback(tmpdir)
+def drop_incompatible_items(d):
+    """Recursively drop incompatible items before assertion"""
+    dd = {}
+    for k, v in d.items():
+        if isinstance(v, dict):
+            dd[k] = drop_incompatible_items(v)
+        elif isinstance(v, (list, set, tuple)):
+            dd[k] = type(v)(drop_incompatible_items(vv) if isinstance(vv, dict) else vv
+                            for vv in v)
+        elif re.match(r'^(?!msg|start|end|delta)', k):
+            dd[k] = v
+    return dd
+
+
+def run_callback(tmpdir, endpoint_type, vcrmode):
+    run = run_playbook_callback(tmpdir, endpoint_type)
     assert run.rc == 0
+    assert len(tmpdir.listdir()) > 0
```

Comment:
*technically* an empty list is falsy, so you could `assert tmpdir.listdir()`, but I am not sure what drives the message "the result directory is not empty" more

Reply:
Okay, let me add a nicer assertion message then.